suite-native/module-send/src/components/AddressReviewStepList.tsx

```diff
@@ -0,0 +1,129 @@
+import { AccountsRootState, DeviceRootState, SendRootState } from '@suite-common/wallet-core';
+import { nativeSpacings } from '@trezor/theme';
+
+import { signTransactionNativeThunk as signTransactionThunk } from '../sendFormThunks';
+import { selectIsFirstTransactionAddressConfirmed } from '../selectors';
+import { SlidingFooterOverlay } from '../components/SlidingFooterOverlay';
+import { AddressReviewStep } from '../components/AddressReviewStep';
+import { CompareAddressHelpButton } from '../components/CompareAddressHelpButton';
+import { AddressOriginHelpButton } from '../components/AddressOriginHelpButton';
+
+const NUMBER_OF_STEPS = 3;
+const OVERLAY_INITIAL_POSITION = 75;
+const LIST_VERTICAL_SPACING = nativeSpacings.medium;
+
+type RouteProps = StackProps<SendStackParamList, SendStackRoutes.SendAddressReview>['route'];
+type NavigationProps = StackNavigationProps<SendStackParamList, SendStackRoutes.SendAddressReview>;
+
+export const AddressReviewStepList = () => {
+    const route = useRoute<RouteProps>();
+    const navigation = useNavigation<NavigationProps>();
+    const { showToast } = useToast();
+    const dispatch = useDispatch();
+
+    const [childHeights, setChildHeights] = useState<number[]>([]);
+    const [stepIndex, setStepIndex] = useState(0);
+
+    const areAllStepsDone = stepIndex === NUMBER_OF_STEPS - 1;
+    const isLayoutReady = childHeights.length === NUMBER_OF_STEPS;
+    const { accountKey, transaction } = route.params;
+
+    const isAddressConfirmed = useSelector(
+        (state: AccountsRootState & DeviceRootState & SendRootState) =>
+            selectIsFirstTransactionAddressConfirmed(state, accountKey),
+    );
+
+    useEffect(() => {
+        if (isAddressConfirmed) {
+            navigation.navigate(SendStackRoutes.SendOutputsReview, { accountKey });
+        }
+    }, [isAddressConfirmed, accountKey, navigation]);
+
+    const handleReadItemListHeight = (event: LayoutChangeEvent, index: number) => {
+        const { height } = event.nativeEvent.layout;
+        setChildHeights(prevHeights => {
+            const newHeights = [...prevHeights];
+            newHeights[index] = height + LIST_VERTICAL_SPACING;
+
+            return newHeights;
+        });
+    };
+
+    const handleNextStep = async () => {
+        setStepIndex(prevStepIndex => prevStepIndex + 1);
+
+        if (stepIndex === NUMBER_OF_STEPS - 2) {
+            const response = await dispatch(
+                signTransactionThunk({
+                    accountKey,
+                    feeLevel: transaction,
+                }),
+            );
+
+            if (isRejected(response)) {
+                // TODO: display error message based on the error code
+                showToast({
+                    variant: 'error',
+                    message: 'Something went wrong',
+                    icon: 'closeCircle',
+                });
+
+                navigation.navigate(SendStackRoutes.SendAccounts);
```

Comment:
why not pop?

Reply:
this is not the previous screen. This is screen at the beginning of the flow. It is basically a restart of the flow from users point of view.

Reply:
fixed 35fb89870f